Add WinHttp cmake option to the contributing guide (#1610)
Azure/azure-sdk-for-cpp: CONTRIBUTING.md

<td>OFF</td>
</tr>
<tr>
<td>BUILD_TRANSPORT_WINHTTP</td>

> <td>Build the WinHTTP transport adapter. When building on Windows systems, if no other transport adapter is built, this option will be automatically turned ON. This option is not supported on non-Windows OSes.</td>

Reviewer: ```suggestion
<td>Build the WinHTTP transport adapter. When building on Windows systems, if no other transport adapter is built, this option is ON by default. This option is not supported on non-Windows OSes.</td>
```

Reviewer: Its a NIT, Up to you if you want to use the suggested wording.

Author: The option isn't actually ON by default. It is turned on by one of our cmake modues, only under certain conditions.
This is similar wording to the curl option above.
https://github.com/Azure/azure-sdk-for-cpp/blob/9ed77d8b0648588432ca3e3c26db911c9041c396/CMakeLists.txt#L13
https://github.com/Azure/azure-sdk-for-cpp/blob/9ed77d8b0648588432ca3e3c26db911c9041c396/cmake-modules/AzureTransportAdapters.cmake#L25-L27
